code/modules/basketball/controller.dm

```diff
@@ -0,0 +1,405 @@
+	var/obj/effect/landmark/basketball/team_spawn/spawn_landmark
+
+	var/team_uniform
+	var/team_name
+
+	// rename the hoops to their appropriate teams names
+	home_hoop.name = current_map.team_name
+	away_hoop.name = away_map.team_name
+
+	var/player_count = 0
+	// if total players is odd number then the odd man out is a referee
+	var/minigame_has_referee = length(ready_players) % 2
+
+	for(var/player_key in ready_players)
+		player_count++
+		minigame_players |= player_key
+
+		var/is_player_referee = (player_count == length(ready_players) && minigame_has_referee)
+
+		if(is_player_referee)
+			spawn_landmark = pick_n_take(referee_spawnpoint)
+			team_uniform = /datum/outfit/basketball/referee
+		else if(player_count % 2) // odd is home team
+			spawn_landmark = pick_n_take(home_spawnpoints)
+			home_team_players |= player_key
+			away_hoop.team_ckeys |= player_key // to restrict scoring on opponents hoop rapidly
+			team_uniform = current_map.home_team_uniform
+			team_name = current_map.team_name
+		else // even is away team
+			spawn_landmark = pick_n_take(away_spawnpoints)
+			away_team_players |= player_key
+			home_hoop.team_ckeys |= player_key // to restrict scoring on opponents hoop rapidly
+			team_uniform = away_map.home_team_uniform
+			team_name = away_map.team_name
+
+		var/mob/living/carbon/human/baller = new(get_turf(spawn_landmark))
+
+		if(baller.dna.species.outfit_important_for_life)
+			baller.set_species(/datum/species/human)
+
+		ADD_TRAIT(baller, TRAIT_NOFIRE, BASKETBALL_MINIGAME_TRAIT)
+		ADD_TRAIT(baller, TRAIT_NOBREATH, BASKETBALL_MINIGAME_TRAIT)
+		ADD_TRAIT(baller, TRAIT_CANNOT_CRYSTALIZE, BASKETBALL_MINIGAME_TRAIT)
+		// this is basketball, not a boxing match
+		ADD_TRAIT(baller, TRAIT_PACIFISM, BASKETBALL_MINIGAME_TRAIT)
+
+		baller.equipOutfit(team_uniform)
+
+		var/client/player_client = GLOB.directory[player_key]
+		if(player_client)
+			player_client.prefs.safe_transfer_prefs_to(baller, is_antag = TRUE)
+		baller.key = player_key
+
+		SEND_SOUND(baller, sound('sound/misc/whistle.ogg', volume=30))
+		if(is_player_referee)
+			to_chat(baller, span_notice("You are a referee. Make sure the teams play fair and use your whistle to call fouls appropriately."))
+		else
+			to_chat(baller, span_notice("You are a basketball player for the [team_name]. Score as much as you can before time runs out."))
+			to_chat(baller, span_info("LMB to pass the ball while on help intent (zero stamina cost/) - accuracy penalty when scoring)"))
+			to_chat(baller, span_info("RMB to shoot the ball ([STAMINA_COST_SHOOTING] stamina cost) - this goes over players heads"))
+			to_chat(baller, span_info("Click directly on hoop while adjacent to dunk ([STAMINA_COST_DUNKING] stamina cost)"))
+			to_chat(baller, span_info("Spinning decreases other players disarm chance against you but reduces shooting accuracy ([STAMINA_COST_SPINNING] stamina cost)"))
+
+/datum/basketball_controller/proc/victory()
```

Comment:
Document this function if you would

Reply:
Doc'd in d5baf59d58aeabad6f2d9e7332e883dbc040435e